src/core_plugins/input_control_vis/public/control/list_control_factory.js

```diff
@@ -70,14 +81,20 @@ class ListControl extends Control {
       this.useTimeFilter,
       ancestorFilters);
 
+    this.lastQuery = query;
     const resp = await searchSource.fetch();
+    if (query && this.lastQuery !== query) {
+      // search results returned out of order - ignore results from old query
+      return;
```

Comment:
Seems this could accidentally blank out the results:

- fetch is called twice ('a', 'b')
- 'b' returns first
- 'a' returns second, and, and fetch returns undefined

The caller of fetch needs to know that if it gets an undefined, it should do nothing (it should keep its current state). Might be better to have this return the correct results if called out of sync (e.g. in the previous example, 'a' would return 'b's results).

Reply:
The return is not used to set anything. The `resp` is used to update `this.selectOptions` later in the function. Calling return early just short circuits that so `this.selectOptions` is not updated with no longer needed data.